realm/realm-library/src/main/java/io/realm/RealmList.java

```diff
@@ -1328,6 +1331,14 @@ private static boolean isClassForRealmModel(Class<?> clazz) {
             //noinspection unchecked
             return (ManagedListOperator<E>) new DateListOperator(realm, osList, (Class<Date>) clazz);
         }
+        if (clazz == Decimal128.class) {
```

Comment:
As discussed, let's add the bson dependency for this PR and optimise later a mechanism to make it optional. I think the method count and size will not affect apps not using these types (because of R8 tree shaking)